### DISCUSSION TITLE
Allow hiding property focus in the Inspector

### DISCUSSION THREAD

**Discussion by mieldepoche** [godotengine/godot-proposals#7628] [Editor]
[simplescreenrecorder-2023-09-08_19.22.28.webm](https://github.com/godotengine/godot-proposals/assets/71937306/d3f36e42-0441-4725-b6c3-ea815d565b2a)

both the value name and input widget can have some sort of focus, and none of them serve a purpose to me.
- name highlight is useless afaik, I think you needed to be able to drag it on visual scripts :thinking: 
- value box highlight at least gives you info on what is focused, but I have no use for it and the visual noise is really distracting, not to mention I find this thick outline really ugly. It makes the editor feel bad and clunky.

Look at how unreal does it:

[simplescreenrecorder-2023-09-08_19.26.10.webm](https://github.com/godotengine/godot-proposals/assets/71937306/b840f539-7420-4f01-8678-a7665115d58b)

clicking outside takes the focus back from the input field. clean.

There are several small things like that that make the editor feel bad to me.

**Comment by Mickeon** [comment]
I recall, and I may be wrong, that Godot 3.x worked similarly, but the switch to Popups, Windows, and the likes of them may have screwed this up, too.

The way Unreal does it seems nice. Hey, the cursor changes to sideway arrows, too. In hindsight that does seem nice...

**Reply by Calinou** [reply]
> The way Unreal does it seems nice. Hey, the cursor changes to sideway arrows, too. In hindsight that does seem nice...

[I have a PR for that](https://github.com/godotengine/godot/pull/81052) 🙂

**Comment by Calinou** [comment]
> - name highlight is useless afaik, I think you needed to be able to drag it on visual scripts 🤔

You can drag-and-drop the name or right-click it to copy its name, path or value.

> - value box highlight at least gives you info on what is focused, but I have no use for it and the visual noise is really distracting, not to mention I find this thick outline really ugly. It makes the editor feel bad and clunky.

The focus outline is here for keyboard navigation, which is necessary for accessibility reasons. What we can do instead is hide the focus outline on mouse/touch-initiated focus, but keep it on keyboard/gamepad-initiated focus (like `:focus-visible` in CSS). This is something I proposed in another proposal, but I can't find it right now.

**Reply by mieldepoche** [reply]
> You can drag-and-drop the name or right-click it to copy its name, path or value.

yes, but that shouldn't require more than a highlight-on-hover, not a whole focusable state.

> What we can do instead...

yes that's a perfect solution, and that's kinda what unreal does too (it allows tabbing through input fields).
